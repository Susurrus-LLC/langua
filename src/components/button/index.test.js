import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Button from './'

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Button />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately with various props and children', () => {
    let tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer.create(
      <Button type='submit' ver='success'>
        Submit
      </Button>
    )
    expect(tree).toMatchSnapshot()

    tree = renderer.create(
      <Button type='reset' ver='danger'>
        Reset
      </Button>
    )
    expect(tree).toMatchSnapshot()

    tree = renderer.create(
      <Button link ver='neutral' route='test'>
        Link
      </Button>
    )
    expect(tree).toMatchSnapshot()

    tree = renderer.create(
      <Button small onClick={jest.fn()}>
        Small Function
      </Button>
    )
  })
})
