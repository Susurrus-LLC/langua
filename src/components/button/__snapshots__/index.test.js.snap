// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders accurately with various props and children 1`] = `
<button
  className="btn"
  role="button"
  type="button"
/>
`;

exports[`Header renders accurately with various props and children 2`] = `
<button
  className="btn success"
  role="button"
  type="submit"
>
  Submit
</button>
`;

exports[`Header renders accurately with various props and children 3`] = `
<button
  className="btn danger"
  role="button"
  type="reset"
>
  Reset
</button>
`;

exports[`Header renders accurately with various props and children 4`] = `
<a
  className="btn neutral buttonLink"
  href="/test"
>
  Link
</a>
`;
